File: 250327/두 가지로 열리는 자물쇠/a-two-way-lock.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split("\n");

const n = Number(input[0]);
const [a, b, c] = input[1].split(" ").map(Number);
const [a2, b2, c2] = input[2].split(" ").map(Number);


function isClose(x, y, n) {
  let diff = Math.abs(x - y);
  let circularDiff = Math.min(diff, n - diff);
  return circularDiff <= 2;
}

let count = 0;

for (let i = 1; i <= n; i++) {
  for (let j = 1; j <= n; j++) {
    for (let k = 1; k <= n; k++) {
      const closeToFirst =
        isClose(i, a, n) && isClose(j, b, n) && isClose(k, c, n);
      const closeToSecond =
        isClose(i, a2, n) && isClose(j, b2, n) && isClose(k, c2, n);

      if (closeToFirst || closeToSecond) {
        count++;
      }
    }
  }
}

console.log(count);

File: 250809/두 가지로 열리는 자물쇠/a-two-way-lock.py
N = int(input())
a1, b1, c1 = map(int, input().split())
a2, b2, c2 = map(int, input().split())

def circ_dist(x, y, N):
    d = abs(x - y)
    return min(d, N - d)  # 원형 거리

cnt = 0
for i in range(1, N + 1):
    for j in range(1, N + 1):
        for k in range(1, N + 1):
            open1 = (
                circ_dist(i, a1, N) <= 2 and
                circ_dist(j, b1, N) <= 2 and
                circ_dist(k, c1, N) <= 2
            )
            open2 = (
                circ_dist(i, a2, N) <= 2 and
                circ_dist(j, b2, N) <= 2 and
                circ_dist(k, c2, N) <= 2
            )
            if open1 or open2:
                cnt += 1

print(cnt)